sui_core/src/authority/authority_store.rs

```diff
@@ -838,9 +838,7 @@ impl<const ALL_OBJ_VER: bool, S: Eq + Serialize + for<'de> Deserialize<'de>>
                 // sequence number (`OBJECT_START_VERSION`). Otherwise use the `scheduled` map to
                 // to assign the next sequence number.
                 let version = v.unwrap_or_else(|| OBJECT_START_VERSION);
-                let next_version = v
-                    .map(|v| v.increment())
-                    .unwrap_or_else(|| SequenceNumber::from(2));
```

Comment:
Could you `unwrap_or_else` a const at `OBJECT_START_VERSION + 1`?

Reply:
we could, but this is even simpler `let next_version = version.increment();`